src/modules/wallet/web3-react/connection/index.tsx

```diff
@@ -25,91 +25,74 @@ import { ledgerConnection, LedgerOption } from './ledger'
 import { networkConnection } from './network'
 import { gnosisSafeConnection } from './safe'
 import { tallyWalletConnection } from './tally'
+import { trezorConnection, TrezorOption } from './trezor'
 import { trustWalletConnection, TrustWalletOption } from './trust'
 import { walletConnectConnection, WalletConnectOption } from './walletConnect'
 import { walletConnectConnectionV2, WalletConnectV2Option } from './walletConnectV2'
 
-// import { ZengoOption } from './zengo'
 import { ConnectionType } from '../../api/types'
 import { Web3ReactConnection } from '../types'
 
-const CONNECTIONS: Web3ReactConnection[] = [
-  gnosisSafeConnection,
-  injectedConnection,
-  coinbaseWalletConnection,
-  walletConnectConnection,
-  walletConnectConnectionV2,
-  fortmaticConnection,
-  networkConnection,
-  tallyWalletConnection,
-  trustWalletConnection,
-  ledgerConnection,
-  keystoneConnection,
-  injectedWidgetConnection,
-]
-
-export function isChainAllowed(connector: Connector, chainId: number) {
-  switch (connector) {
-    case fortmaticConnection.connector:
-      return chainId === SupportedChainId.MAINNET
-    case injectedConnection.connector:
-    case coinbaseWalletConnection.connector:
-    case walletConnectConnection.connector:
-    case networkConnection.connector:
-    case gnosisSafeConnection.connector:
-    case tallyWalletConnection.connector:
-    case trustWalletConnection.connector:
-    case injectedWidgetConnection.connector:
-    case ledgerConnection.connector:
-    case keystoneConnection.connector:
-    case walletConnectConnectionV2.connector:
-      return ALL_SUPPORTED_CHAIN_IDS.includes(chainId)
-    default:
-      return false
-  }
+const allowedChainsByWallet: Record<ConnectionType, SupportedChainId[]> = {
+  [ConnectionType.FORTMATIC]: [SupportedChainId.MAINNET],
+  [ConnectionType.INJECTED]: ALL_SUPPORTED_CHAIN_IDS,
+  [ConnectionType.INJECTED_WIDGET]: ALL_SUPPORTED_CHAIN_IDS,
+  [ConnectionType.COINBASE_WALLET]: ALL_SUPPORTED_CHAIN_IDS,
+  [ConnectionType.WALLET_CONNECT]: ALL_SUPPORTED_CHAIN_IDS,
+  [ConnectionType.WALLET_CONNECT_V2]: ALL_SUPPORTED_CHAIN_IDS,
+  [ConnectionType.NETWORK]: ALL_SUPPORTED_CHAIN_IDS,
+  [ConnectionType.GNOSIS_SAFE]: ALL_SUPPORTED_CHAIN_IDS,
+  [ConnectionType.TALLY]: ALL_SUPPORTED_CHAIN_IDS,
+  [ConnectionType.TRUST]: ALL_SUPPORTED_CHAIN_IDS,
+  [ConnectionType.LEDGER]: ALL_SUPPORTED_CHAIN_IDS,
+  [ConnectionType.TREZOR]: ALL_SUPPORTED_CHAIN_IDS,
+  [ConnectionType.KEYSTONE]: ALL_SUPPORTED_CHAIN_IDS,
+  [ConnectionType.ALPHA]: [],
+  [ConnectionType.AMBIRE]: [],
+  [ConnectionType.AMBIRE]: [],
+  [ConnectionType.ZENGO]: [],
```

Comment:
Does this mean these wallets have no supported chains?

Reply:
I haven't tested it, just inherited logic from the previous code.
Will check it

Reply:
<img width="575" alt="image" src="https://github.com/cowprotocol/cowswap/assets/7122625/71bd7bdf-a661-4150-96d2-01dd5b9a5cf0">

Yes, before there was the same logic